ingress/controllers/nginx/controller.go

```diff
@@ -217,6 +257,75 @@ func (lbc *loadBalancerController) getUDPConfigMap(ns, name string) (*api.Config
 	return lbc.client.ConfigMaps(ns).Get(name)
 }
 
+func (lbc *loadBalancerController) updateEpNamedPorts(key string) {
+	if !lbc.controllersInSync() {
+		lbc.svcEpQueue.requeue(key, fmt.Errorf("deferring sync till endpoints controller has synced"))
+		return
+	}
+
+	svcObj, svcExists, err := lbc.svcLister.GetByKey(key)
+	if err != nil {
+		glog.Warningf("error getting service %v: %v", key, err)
+		return
+	}
+
+	if !svcExists {
+		glog.Warningf("service %v not found", key)
+		return
+	}
+
+	svc := svcObj.(*api.Service)
+	if svc.Spec.Selector == nil {
+		return
+	}
+
+	pods, err := lbc.client.Pods(svc.Namespace).List(api.ListOptions{
```

Comment:
> I'd suggest you don't actually list all pods

this just get the pods associated with one service, not all the pods in the cluster

> currently it happens every svc enqueue right?

I changed that to Ingress add and update events.